pkg/providerconfig/types/types.go

```diff
@@ -226,7 +227,11 @@ func (configVarString *ConfigVarString) UnmarshalJSON(b []byte) error {
 	if !bytes.HasPrefix(b, []byte("{")) {
 		b = bytes.TrimPrefix(b, []byte(`"`))
 		b = bytes.TrimSuffix(b, []byte(`"`))
-		configVarString.Value = string(b)
+		s, err := strconv.Unquote("\"" + string(b) + "\"")
```

Comment:
We should leave a comment that we intentionally remove prefix and suffix quotes and then add them for `Unquote`. That's because we don't know are there any quotes before calling `TrimPrefix` and `TrimSuffix`, and if the original byte slice doesn't have them, `Unquote` will fail. This is ensuring that this doesn't happen, i.e. we explicitly quote string, so `Unquote` will never fail.

Reply:
Updated